src/main/java/io/prometheus/cloudwatch/CloudWatchCollector.java

```diff
@@ -216,8 +221,15 @@ public String getMonitoringEndpoint(Region region) {
       String nextToken = null;
       do {
         request.setNextToken(nextToken);
-        ListMetricsResult result = client.listMetrics(request);
-        cloudwatchRequests.inc();
+        ListMetricsResult result;
+        try {
+          result = client.listMetrics(request);
+          cloudwatchRequests.inc();
+        } catch (AmazonCloudWatchException ex) {
+          cloudwatchErrorRequests.labels(ex.getErrorCode(), "listMetrics").inc();
```

Comment:
How many values can the error code have?

Reply:
It's hard to definitively say as it's not well documented. Based on looking at error codes for a variety of AWS services, the possible values are limited do not include "unique" data that'd cause the cardinality to balloon (unique data, such as an instance id, appears in the 'message' field).

The most codes I've run across is for EC2 (https://docs.aws.amazon.com/AWSEC2/latest/APIReference/errors-overview.html), if that gives any sense of worst-case distinct values for the error code.

The error values themselves are nice to have, but not a necessity.  It just means that you'd have to take an extra step and search the `cloudwatch_exporter` logs to determine the underlying cause.

Reply:
That's a lot of error codes in the worst case. How many have you seen in practice?

You'll already know if the scrape failed due to `up` being 0.

Reply:
I've seen just a handful of error codes in practice.

AWS is pushing back on increasing the CloudWatch limits, so I'm going to need to see if I can determine what all else is querying CloudWatch on my side and possibly tune that down.

In the meantime I can monitor `cloudwatch_requests_total` to see if it stops increasing to know if we're getting heavily rate limited.